File: python/7_Reverse Integer.py
# 7. Reverse Integer
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        flag = 1
        if x < 0:
            flag = -1
            x = -x
        x = str(x)
        x = x[::-1]
        x = int(x)
        if (flag == 1) and (x < 0):
            return 0
        if x > 2147483648:
            return 0
        return flag*x

File: java/7_Reverse_Integer.java
public class Solution {
    public int reverse(int x) {
        int sign = 1, result = 0;
        if (x < 0){
            sign = -1;
        }
        x = Math.abs(x);
        if (x < 0){
            return 0;
        }
        while (x != 0){
            int tail = x % 10;
            if ( result > Integer.MAX_VALUE / 10){
                return 0;
            }
            if (result == Integer.MAX_VALUE / 10 && sign == 1 && tail > Integer.MAX_VALUE % 10){
                return 0;
            }
            if (result == Integer.MAX_VALUE / 10 && sign == -1 && tail > Math.abs(Integer.MIN_VALUE + 10 ) % 10){
                return 0;
            }
            result = result * 10 + tail;
            x = x / 10;
        }
        return result * sign;
    }
}